packages/core/test/integration/query-interface/bulk-delete.test.ts

```diff
@@ -0,0 +1,105 @@
+import { expect } from 'chai';
+import { DataTypes } from '@sequelize/core';
+import { sequelize } from '../support';
+
+const queryInterface = sequelize.queryInterface;
+
+describe('QueryInterface#bulkDelete', () => {
+  beforeEach(async () => {
+    await queryInterface.createTable('level', {
+      id: {
+        type: DataTypes.INTEGER,
+        primaryKey: true,
+        autoIncrement: true,
+      },
+      name: {
+        type: DataTypes.STRING,
+        allowNull: false,
+      },
+    });
+
+    await queryInterface.createTable('actors', {
+      id: {
+        type: DataTypes.INTEGER,
+        primaryKey: true,
+        autoIncrement: true,
+      },
+      name: {
+        type: DataTypes.STRING,
+        allowNull: false,
+      },
+      status: {
+        type: DataTypes.STRING,
+        allowNull: true,
+      },
+      age: {
+        type: DataTypes.INTEGER,
+        allowNull: false,
+      },
+      level_id: {
+        type: DataTypes.INTEGER,
+        allowNull: false,
+        references: {
+          key: 'id',
+          table: 'level',
+        },
+        onDelete: 'CASCADE',
+        onUpdate: 'CASCADE',
+      },
+    });
+
+    await queryInterface.bulkInsert('level', [
+      { name: 'level1' },
+      { name: 'level2' },
+      { name: 'level3' },
+    ]);
+
+    await queryInterface.bulkInsert('actors', [
+      { name: 'actor1', status: 'active', age: 10, level_id: 1 },
+      { name: 'actor2', status: 'active', age: 20, level_id: 2 },
+      { name: 'actor3', status: 'active', age: 30, level_id: 3 },
+    ]);
+  });
+
+  it('should truncate the table', async () => {
+    await queryInterface.bulkDelete('actors', {}, { truncate: true });
+
+    const count = await queryInterface.select(null, 'actors', {});
+    expect(count).to.have.length(0);
+  });
+
+  if (sequelize.dialect.supports.truncate.cascade) {
+    it('should truncate the table with cascade', async () => {
```

Comment:
It's a small performance improvement, but can we make this a new describe block with a `beforeEach` to create the level table and the link to actors?

Reply:
Done